### DISCUSSION TITLE
zod url fails when having port number and field/value pairs

### DISCUSSION THREAD

**Discussion by ricardo-valero** [colinhacks/zod#1479] [Q&A] [answered]
This url works fine
`"sqlserver://database-test.amazonaws.com;database=x;user=y;password=z;trustServerCertificate=true"`

But this doesn't
`"sqlserver://192.168.100.10:1433;database=x;user=y;password=z;trustServerCertificate=true"`

**Comment by JacobWeisenburger** [comment]
Confirmed

```ts
const schema = z.string().url()

console.log(
    schema.safeParse(
        'sqlserver://database-test.amazonaws.com;database=x;user=y;password=z;trustServerCertificate=true'
    ).success
) // true

console.log(
    schema.safeParse(
        'sqlserver://192.168.100.10:1433;database=x;user=y;password=z;trustServerCertificate=true'
    ).success
) // false
```

**Comment by JacobWeisenburger** [comment, marked as answer]
It seems that this is working as intended

please see [this issue](https://github.com/colinhacks/zod/issues/1687) for more info
